jwst/lib/set_telescope_pointing.py

```diff
@@ -1452,6 +1452,9 @@ def populate_model_from_siaf(model, siaf):
         model.meta.wcsinfo.cdelt1 = siaf.cdelt1 / 3600  # in deg
         model.meta.wcsinfo.cdelt2 = siaf.cdelt2 / 3600  # in deg
         model.meta.coordinates.reference_frame = "ICRS"
+    elif model.meta.exposure.type.lower() == 'nrc_tsgrism':
+        model.meta.wcsinfo.siaf_xref_sci = siaf.crpix1
+        model.meta.wcsinfo.siaf_yref_sci = siaf.crpix2
```

Comment:
It might be worth adding both of these to the `wcsinfo.schema` so they are checked when read in and written out.  Just don't give it a `fits_keyword` or `fits_hdu` attribute.  So something like:
```
          siaf_xref_sci:
            title: Aperture X reference point in SCI frame from SIAF
            type: number
          siaf_yref_sci:
            title: Aperture Y reference point in SCI frame from SIAF
            type: number
```
?

It will make it clear that they're supposed to be there.